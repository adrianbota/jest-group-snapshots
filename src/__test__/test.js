import group from '../index.js';

test('Jest Group Snapshots', () => {
    const g = group();

    const before1 = document.querySelector('body').cloneNode(false);
    before1.style.color = 'red';
    g.add(before1, 'Body element before');

    const after1 = document.querySelector('body').cloneNode(false);
    after1.style.color = 'blue';
    g.add(after1, 'Body element after');

    expect(g.values()).toMatchSnapshot('ELEMENT');

    expect(group()
        .add({}, 'Flag disabled')
        .add({ enabled: true }, 'Flag enabled')
        .values()
    ).toMatchSnapshot('Object');
});
