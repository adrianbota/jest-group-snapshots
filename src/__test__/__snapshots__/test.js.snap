// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Group Snapshots: ELEMENT 1`] = `
*** Body element before ***
<body
  style="color: red;"
/>

*** Body element after ***
<body
  style="color: blue;"
/>
`;

exports[`Jest Group Snapshots: Object 1`] = `
*** Flag disabled ***
Object {}

*** Flag enabled ***
Object {
  "enabled": true,
}
`;
